docs/FAQs.md

```diff
@@ -13,6 +13,109 @@ After you set up a repository, you can substitute in any tools you'd like.
+   - update `.github/workflows/build.yml` to ensure `dist` is up to date:
+
+     <details>
+         <summary><code>.github/workflows/build.yml</code></summary>
+
+     ```yml
+     jobs:
+       build:
+         runs-on: ubuntu-latest
+         steps:
+           - uses: actions/checkout@v4
+           - uses: ./.github/actions/prepare
+           - run: pnpm build
+
+           - name: Compare dist/index.js
+             run: |
+               if [ "$(git diff --ignore-space-at-eol --text dist/index.js | wc -l)" -gt "0" ]; then
+                 echo "Detected uncommitted changes after build."
+                 echo "You may need to run 'pnpm run build' locally and commit the changes."
+                 echo ""
+                 echo "See diff below:"
+                 echo ""
+                 git diff --ignore-space-at-eol --text dist/index.js
+                 echo ""
+                 # say this again in case the diff is long
+                 echo "You may need to run 'pnpm run build' locally and commit the changes."
+                 echo ""
+                 exit 1
+               fi
+
+           - id: upload
+             if: ${{ failure() && steps.diff.outcome == 'failure' }}
+             name: Upload artifact
+             uses: actions/upload-artifact@v4
+             with:
+               name: dist
+               path: dist/
+
+     name: Build
+
+     on:
+       pull_request: ~
+       push:
+         branches:
+           - main
+
+     permissions:
+       contents: read
+     ```
+
+      </details>
+
+   - GitHub Actions run without installing package dependencies.
+     Replace `tsup` with [`ncc`](https://github.com/vercel/ncc) to build source files and dependencies into a single JS file.
+     Delete `tsup.config.ts` then execute the following commands:
+
+   ```bash
+   pnpm remove tsup
+   pnpm add @vercel/ncc -D
+   pnpm add @actions/core -P
+   ```
+
+   - We also added the [`@actions/core`](https://github.com/actions/toolkit/tree/master/packages/core) package, used for building GitHub Actions.
+     Now we need to update the `build` script in our `package.json`:
+
+   ```diff
+   -"build": "tsup",
+   +"build": "ncc build src/index.ts -o dist --license licenses.txt",
+   ```
+
+   - Our build now emits to the `dist` directory; so we'll want to avoid linting that directory by adding the following to `.eslintignore` and our `.prettierignore`:
+
+   ```diff
+   +dist
+   ```
+
+   - Rather than having to remember to compile each time, we'll update our precommit hook in `.husky/precommit` to build for us on each commit:
+
+   ```diff
+   +pnpm run build
+   +git add dist
+   npx lint-staged
+   ```
+
+2. We're going to need an [`action.yml`](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions) file - [here's an example](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#creating-an-action-metadata-file).
```

Comment:
[Proofreading] (re-creating https://github.com/JoshuaKGoldberg/create-typescript-app/pull/1196#discussion_r1443853189) The second linked page links to the first linked page itself. I think we can consolidate this item a bit. How about:

```suggestion
3. Create an [`action.yml` metadata file](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#creating-an-action-metadata-file).
```